assets/src/scripts/store/index.js

```diff
@@ -104,6 +105,48 @@ export const Actions = {
             );
         };
     },
+    retrieveExtendedTimeseries(site, period) {
+        return function(dispatch, getState) {
+            const state = getState();
+            const parmCd = getCurrentParmCd(state);
+            const tsKey = period === 'P7D' ? 'current' : `current:${period}:${parmCd}`;
```

Comment:
It'd be nice to keep the key format isolated to one spot - maybe this could be retrieved via a function call, or hasTimeseries() could take the parts in?

Reply:
Agreed. I look at reworking this.